lib/javaHapiValidator.ts

```diff
@@ -18,10 +23,47 @@ export default class JavaHapiValidator extends Stack {
     constructor(scope: Construct, id: string, props: JavaHapiValidatorProps) {
         super(scope, id, props);
 
-        this.hapiValidatorLambda = new Function(scope, 'validator', {
+        const igEncryptionKey = new Key(scope, `igEncryptionKey-${props.stage}`, {
+            enableKeyRotation: true,
+            description: 'KMS CMK for IGs',
+            policy: new PolicyDocument({
+                statements: [
+                    new PolicyStatement({
+                        sid: 'Enable IAM Root Permissions',
+                        effect: Effect.ALLOW,
+                        actions: ['kms:*'],
+                        resources: ['*'],
+                        principals: [new AnyPrincipal()],
```

Comment:
Should this be `AccountRootPrincipal()`?

Reply:
This sadly doesn't work with the AccountRootPrincipal, no matter what permission set I tried. Runs into an error when trying to decrypt due to access denied.

Reply:
Just kidding, I had just missed a permission on the lambda itself. Updated!